src/d/actor/d_a_npc_bs1.cpp

```diff
@@ -137,161 +567,651 @@ void daNpc_Bs1_c::shop_talk() {
+            } else {
+                m740 = 0x2F53;
+            }
+            dComIfGp_event_reset();
+            if (m770 != &daNpc_Bs1_c::wait_action) {
+                if (m770) {
+                    m834 = -1;
+                    (this->*m770)(NULL);
+                }
+                m770 = &daNpc_Bs1_c::wait_action;
+                m834 = 0;
+                (this->*m770)(NULL);
+            }
+        }
+        lookBack();
+    }
+    return TRUE;
 }
 
 /* 00003CB4-00003D7C       .text evn_talk_init__11daNpc_Bs1_cFi */
-void daNpc_Bs1_c::evn_talk_init(int) {
+BOOL daNpc_Bs1_c::evn_talk_init(int actorId) {
     /* Nonmatching */
+    u32* pMsgNo = dComIfGp_evmng_getMyIntegerP(actorId, "MsgNo");
+    u32* pEndMsgNo = dComIfGp_evmng_getMyIntegerP(actorId, "EndMsgNo");
+    l_msgId = 0xFFFFFFFF;
+    l_msg = NULL;
+    if (pMsgNo != NULL) {
+        m738 = *pMsgNo;
+    } else {
+        m738 = 0;
+    }
+    if (pEndMsgNo != NULL) {
+        m744 = *pEndMsgNo;
+    } else {
+        m744 = 0;
+    }
+    return TRUE;
 }
 
 /* 00003D7C-00003DE4       .text evn_continue_talk_init__11daNpc_Bs1_cFi */
-void daNpc_Bs1_c::evn_continue_talk_init(int) {
+BOOL daNpc_Bs1_c::evn_continue_talk_init(int actorId) {
     /* Nonmatching */
+    u32* pEndMsgNo = dComIfGp_evmng_getMyIntegerP(actorId, "EndMsgNo");
+    if (pEndMsgNo != NULL) {
+        m744 = *pEndMsgNo;
+    } else {
+        m744 = 0;
+    }
+    return TRUE;
 }
 
 /* 00003DE4-00003F14       .text evn_talk__11daNpc_Bs1_cFv */
-void daNpc_Bs1_c::evn_talk() {
-    /* Nonmatching */
+BOOL daNpc_Bs1_c::evn_talk() {
+    if (l_msgId == 0xFFFFFFFF) {
+        l_msgId = fopMsgM_messageSet(m738, &eyePos);
+    } else if (l_msg == NULL) {
+        l_msg = fopMsgM_SearchByID(l_msgId);
+        fopMsgM_demoMsgFlagOn();
+    } else {
+        setAnmFromMsgTag();
+        u16 msgStatus = l_msg->mStatus;
+        if (msgStatus == 0xE) {
+            l_msg->mStatus = next_msgStatus(&m738);
+            if (l_msg->mStatus == 0xF) {
+                fopMsgM_messageSet(m738);
+            }
+        } else {
+            if (msgStatus == 0x12) {
+                l_msg->mStatus = 0x13;
+                l_msg = NULL;
+                l_msgId = 0xFFFFFFFF;
+                return TRUE;
+            }
+            if ((msgStatus == 2 || msgStatus == 6) && (m738 == m744)) {
+                m744 = 0;
+                return TRUE;
+            }
+        }
+    }
+    return FALSE;
 }
 
 /* 00003F14-00003FE0       .text evn_jnt_lock_init__11daNpc_Bs1_cFi */
-void daNpc_Bs1_c::evn_jnt_lock_init(int) {
+BOOL daNpc_Bs1_c::evn_jnt_lock_init(int actorIdx) {
     /* Nonmatching */
+    u32* substance = dComIfGp_evmng_getMyIntegerP(actorIdx, "prm");
+    u32 jnt_to_lock;
+    if (substance != NULL) {
+        jnt_to_lock = *substance;
+    } else {
+        jnt_to_lock = 0;
+    }
+    switch (jnt_to_lock) {
+    case 0:
+        mJntCtrl.mbHeadLock = FALSE;
+        mJntCtrl.mbBackBoneLock = FALSE;
```

Comment:
```suggestion
        mJntCtrl.offHeadLock();
        mJntCtrl.offBackBoneLock();
```
etc for the other cases